woocommerce/google-listings-and-ads
# Update phpunit to version 9.5

### Changes proposed in this Pull Request:

This PR updates the phpunit version to 9.5 so we no longer need a workaround to run unit tests for PHP 8. Additionally the default tests for all WordPress versions (L-2) are run on PHP 8. For PHP 7.4 we only run a test with the latest WP version.

Resolves part of #1758 

### Package upgrades
`composer require phpunit/phpunit "^9.5" --dev --update-with-all-dependencies`

```
- Removing phpdocumentor/reflection-common (2.2.0)
- Removing phpdocumentor/reflection-docblock (5.3.0)
- Removing phpdocumentor/type-resolver (1.6.1)
- Removing phpspec/prophecy (v1.15.0)
- Removing phpunit/php-token-stream (3.1.3)
- Removing webmozart/assert (1.11.0)
- Upgrading doctrine/instantiator (1.4.1 => 1.5.0)
- Locking nikic/php-parser (v4.15.3)
- Upgrading phar-io/manifest (1.0.3 => 2.0.3)
- Upgrading phar-io/version (2.0.1 => 3.2.1)
- Upgrading phpunit/php-code-coverage (6.1.4 => 9.2.24)
- Upgrading phpunit/php-file-iterator (2.0.5 => 3.0.6)
- Locking phpunit/php-invoker (3.1.1)
- Upgrading phpunit/php-text-template (1.2.1 => 2.0.4)
- Upgrading phpunit/php-timer (2.1.3 => 5.0.3)
- Upgrading phpunit/phpunit (7.5.20 => 9.5.28)
- Locking sebastian/cli-parser (1.0.1)
- Locking sebastian/code-unit (1.0.8)
- Upgrading sebastian/code-unit-reverse-lookup (1.0.2 => 2.0.3)
- Upgrading sebastian/comparator (3.0.3 => 4.0.8)
- Locking sebastian/complexity (2.0.2)
- Upgrading sebastian/diff (3.0.3 => 4.0.4)
- Upgrading sebastian/environment (4.2.4 => 5.1.4)
- Upgrading sebastian/exporter (3.1.4 => 4.0.5)
- Upgrading sebastian/global-state (2.0.0 => 5.0.5)
- Locking sebastian/lines-of-code (1.0.3)
- Upgrading sebastian/object-enumerator (3.0.4 => 4.0.4)
- Upgrading sebastian/object-reflector (1.1.2 => 2.0.4)
- Upgrading sebastian/recursion-context (3.0.1 => 4.0.4)
- Upgrading sebastian/resource-operations (2.0.2 => 3.0.3)
- Locking sebastian/type (3.2.0)
- Upgrading sebastian/version (2.0.1 => 3.0.2)
- Upgrading symfony/polyfill-ctype (v1.26.0 => v1.27.0)
```

`composer update wp-cli/i18n-command --with-all-dependencies`

```
- Upgrading gettext/gettext (v4.8.6 => v4.8.8)
- Upgrading gettext/languages (2.9.0 => 2.10.0)
- Upgrading mck89/peast (v1.14.0 => v1.15.1)
- Upgrading mustache/mustache (v2.14.1 => v2.14.2)
- Upgrading symfony/finder (v5.4.8 => v5.4.19)
- Upgrading symfony/polyfill-php80 (v1.26.0 => v1.27.0)
- Upgrading wp-cli/i18n-command (v2.3.0 => v2.4.1)
- Upgrading wp-cli/php-cli-tools (v0.11.13 => v0.11.17)
- Upgrading wp-cli/wp-cli (v2.6.0 => v2.7.1)
```

### Detailed test instructions:
1. Checkout this branch 
2. Re-run `composer install` to update dependencies
3. Setup tests: `bin/install-wp-tests.sh <db_name> <db_user> <db_pass>`
4. Run the tests locally (on PHP 8): `vendor/bin/phpunit`

Check the automated tests in this PR and ensure they all pass. We expect to run the following combination of tests:
- WP latest PHP 7.4
- WP 5.9 PHP 8.0
- WP 6.0 PHP 8.0
- WP latest PHP 8.0

Since we updated the WP-CLI packages (only used for development tools), we can confirm those are working by running:
`nvm use && npm install && npm run build`

### Additional details:

I also tried running the unit tests on PHP 8.1 which still produces some deprecation warnings as [mentioned here](https://github.com/woocommerce/google-listings-and-ads/issues/1758#issuecomment-1310294493).
Testing with PHP 8.2 almost all the tests fail, so we'll need some additional investigation to get that running.

### Changelog entry
* Dev - Update phpunit to version 9.5.
